ral-facilities/datagateway-api
# Feature/schema references #116

# Description
Changes the swagger generator to put responses in a separate section and reference them from the endpoint. For example, the responses go in:
```YAML
components:
  schemas:
    Datafile:
      properties:
        CREATE_ID:
          type: string
...
```
and the reference looks like:
```YAML
...
responses:
  '200':
    content:
      application/json:
        schema:
          $ref: '#/components/schemas/Datafile'
...
```
Documentation on swagger is [here](https://swagger.io/docs/specification/describing-responses/#body)

# Testing
1. in a clone of the repository, edit config.json setting `"generate_swagger": true, ...`.
2. run `python3 -m src.main`
3. press `CTRL-C` to stop the server
4. inspect the file at `src/swagger/openapi.yaml`
5. upload it to https://editor.swagger.io to verify the syntax

# Agile Board Tracking
Closes #116
